common/tasks/fifo_scheduler.go

```diff
@@ -182,6 +182,10 @@ func (f *FIFOScheduler[T]) startWorkers(
 func (f *FIFOScheduler[T]) stopWorkers(
 	count int,
 ) {
+	if count > len(f.workerShutdownCh) {
```

Comment:
I guess I will validate `targetWorkerNum := f.options.WorkerCount()` is positive instead. and don't change the worker count if the value is invalid.

Reply:
Reverted the changes to stopWorkers() method and moved the check to workerMonitor(). This works too. But it's more difficult to test. This methods depends on 
- A 1 minute timer which is hard to override. And it's not a good idea to wait for timers in tests.
- A dynamic config read. This can be mocked I think.

Please let me know if there is a better way to test this. If not, I'm ok skipping the tests as well.

Reply:
I can think of a couple ways like extracting the logic after timer is fired to a separate function and test that. Or make the timer duration a config to the Scheduler. 
Re. dynamic config, I think in there's a test implementation that allows you to override config at runtime, probably under `tests`.

But ^ requires some refactoring work and I think the real issue is still the general config validation itself, so I am so with skipping the tests here.